index.js

```diff
@@ -24,17 +24,77 @@ module.exports = {
         if (hasRecps(input.content)) {
           return cb(new Error('recps-guard: should not have recps && allowPublic, check your code'))
         }
-        return publish(input.content, cb)
+
+        if (keepOptions) {
+          return publish({...input.content, options: input.options}, cb)
+        } else {
+          return publish(input.content, cb)
+        }
       }
 
       cb(new Error(`recps-guard: public messages of type "${input.type}" not allowed`))
-    })
+    }
+
+    if (ssb.publish) {
+      ssb.publish.hook(publishHook(false))
+
+      ssb.publish.hook = () => {
+        throw new Error('ssb-recps-guard must be the last to hook ssb.publish')
+        // NOTE because of the last hook get run first we need to guarentee
+        // that no other hooks on publish occured after our, otherwise we cannot
+        // guarentee other hooks do not bypass the guard
+      }
+    }
 
-    ssb.publish.hook = () => {
-      throw new Error('ssb-recps-guard must be the last to hook ssb.publish')
-      // NOTE because of the last hook get run first we need to guarentee
-      // that no other hooks on publish occured after our, otherwise we cannot
-      // guarentee other hooks do not bypass the guard
+    if (ssb.tribes && ssb.tribes.publish) {
+      ssb.tribes.publish.hook(publishHook(true))
+
+      ssb.tribes.publish.hook = () => {
+        throw new Error('ssb-recps-guard must be the last to hook ssb.tribes.publish')
+        // NOTE because of the last hook get run first we need to guarentee
+        // that no other hooks on publish occured after our, otherwise we cannot
+        // guarentee other hooks do not bypass the guard
+      }
+    }
+
+    if (ssb.db && ssb.db.create) {
+      ssb.db.create.hook((create, args) => {
+        const [input, cb] = args
+
+        if (
+          input.encryptionFormat !== undefined ||
+          isEncrypted(input.content) ||
+          (hasRecps(input.content) && !isAllowPublic2(input)) ||
+          allowedTypes.has(input.content.type)
+        ) {
+          return create(input, cb)
+        }
+
+        if (isAllowPublic2(input)) {
+          if (hasRecps(input.content)) {
+            return cb(new Error('recps-guard: should not have recps && allowPublic, check your code'))
+          }
+
+          if (input.content.options) {
+            delete input.content.options.allowPublic
+
+            if (Object.keys(input.content.options).length === 0) {
+              delete input.content.options
+            }
+          }
+
+          return create(input, cb)
```

Comment:
:fire: there are maybe some gotchas here. 

I think you need to use this pattern to be use you don't drop `this` context of `create`

```js
ssb.db.create.hook(function (create, args) {
  
  create.apply(this, args)
})
```

Reply:
:fire: same will be true for the other hook

Reply:
note `function () {}` has `this` scope ` () => {}` does not

Reply:
maybe, but the code worked like this before this PR. for another PR?

Reply:
true... honestly I get confused/ scared about `this` so avoid it where possible.
I think the important thing is that you need `create` to have the context of the object it was initially attached to. So you need an `create.apply` or `create.call` or `create.bind` (?), where you get the `this` from is another matter :P

Reply:
i'd prefer to not do this in this PR since it feels unrelated but i guess it's also small

Reply:
i don't fully understand this but will the code you posted above help? the `this` it's referring to would be the createHook itself right? wouldn't it need to actually point to ssb.db or ssb.db.create?

Reply:
fwiw create() doesn't seem to use `this` https://github.com/ssbc/ssb-db2/blob/83927c0d3de5f4d3623ad0e1c85c9285ae47f33b/core.js#L642